### DISCUSSION TITLE
Installing Error On Android Platform

### DISCUSSION THREAD

**Discussion by dipeshSam** [flet-dev/flet#4242] [Q&A]
I'm trying to run my quick tests on my [PyDroid](https://play.google.com/store/apps/details?id=ru.iiec.pydroid3) mobile app. But following error is coming up:
![Screenshot_2024_1027_143944.jpg](https://github.com/user-attachments/assets/99d50302-9c32-4eb1-8d33-2a5a0bcaa453)

Any kind of help would be appreciated.
Regards.

**Comment by LegendaryPistachio** [comment]
根据你提供的错误信息，问题主要是由于缺少 Rust 编译器导致的。以下是解决此问题的步骤：

1. **安装 Rust 编译器**：
   - 如果你还没有安装 Rust，可以通过以下命令安装：
     ```bash
     curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
     ```
   - 安装完成后，确保将 Rust 的 `bin` 目录添加到你的 `PATH` 中。通常这个目录是 `~/.cargo/bin`。

2. **更新 pip**：
   - 确保你的 pip 是最新版本。可以使用以下命令更新：
     ```bash
     pip install --upgrade pip
     ```

3. **重新安装包**：
   - 在安装 Rust 和更新 pip 后，尝试重新安装你之前要安装的包。

4. **使用预编译的 wheel**：
   - 如果可能，查找该包是否有预编译的 wheel 文件，这样可以避免需要编译的步骤。

5. **检查依赖**：
   - 确保所有依赖项都已正确安装，并且没有其他缺失的编译工具。

**Reply by dipeshSam** [reply]
@cqkxg Thank you for your support.
I tried the first command and got the following output:
```
/storage/emulated/0 $ curl
curl: try 'curl --help' or 'curl --manual' for more information
/storage/emulated/0 $ curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
info: downloading installer
Warning: Not enforcing strong cipher suites for TLS, this is potentially less secure
error: "/data/user/0/ru.iiec.pydroid3/cache/tmp.VP6rvEwlHK/rustup-init": executable's TLS segment is underaligned: alignment is 16, needs to be at least 64 for ARM64 Bionic
Aborted
/storage/emulated/0 $
```

**Any openion on this?**

Thank you and regards.
